python/hail/expr/expression.py

```diff
@@ -19,11 +19,11 @@ def to_expr(e):
         return Expression(ClassMethod('toInt64', Literal('"{}"'.format(e))), TInt64())
     elif isinstance(e, float):
         return Expression(Literal(str(e)), TFloat64())
-    elif isinstance(e, Variant):
+    elif isinstance(e, hail.genetics.Variant):
```

Comment:
You can't load these in the header?  I maybe start to understand your comment about circular imports.  What's the rule here to make sense of how the imports are organized?

Reply:
The way I'd handle the circularity here is to have a list of conversion functions that can be registered, and then have the genetics module register a conversion function for genetics classes.  All the default (built in) stuff can be in a conversion function defined here.  That way, expr doesn't have to import genetics and you break the dependence.

Reply:
this is a good idea. I'd like to save this for a future PR though.

Reply:
I can replace all these with a header import, though.

Reply:
I also didn't totally understand how it all worked until recently, and didn't fully clean up.